doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -558,17 +572,18 @@ file is a particular case of a module called a *library file*.
    allowed for `-Q` only if `From` is present, that is if a prefix is
    given.
 
-   For instance, ``-Q path Lib`` associates the file
-   ``path/Foo/File.vo`` with the logical name
-   ``Lib.Foo.File`` but allows this file to be accessed through
-   the name ``Lib.Foo.File`` when no `From` is given, with the
-   names ``Foo.File`` and ``File`` when :n:`From
-   Lib` is given, and with the name ``File`` when
-   :n:`From Lib.Foo` is given. Additionally, ``-R path Lib`` allows the
-   same file to be accessed through the names
-   ``Foo.File`` and ``File`` when no `From` is given. In particular,
-   `From` is useful to ensure that the file comes from a particular
-   package or subpackage.
+   Note that if the same logical name is given by multiple `-R` or `-Q` options,
+   the last one specified is used.  The output of :cmd:`Print LoadPath` will
+   show both options in last to first order (i.e., `Require` uses the first match
+   from the top).
+
+   Also note that packages installed in `user-contrib` or accessed through `COQPATH` are
+   made available as if with a `-Q` option.  If you create a project outside of these
+   using `-R` in `_CoqProject` and then later put in one of these locations,
+   :cmd:`Require` commands in your code that depend on the `-R` behavior will fail in
+   the new location.  Therefore we recommend using only `-Q` and not `-R`.  In contrast,
```

Comment:
Do we really recommend that?

Reply:
I agree that we don't recommend that. In fact, moving one part of your project to `user-contrib` and keeping the rest somewhere else is what should be discouraged. Only use `-R` to refer to what's inside your library, that you will install at once in `user-contrib`.

Reply:
I missed updating this.  I'll say something similar to @Zimmi48's wording, e.g. "Only use `-R` to refer to files within a single package.  Use `-Q` to refer to files in other packages".

Also, the following may also be incorrect--do you guys know?  Are partially qualified names resolved at compile time or run time?  My guess is that it's at compile time (in which case this text is ~100% incorrect); resolving PQ names at run time seems like it could lead to obscure errors.

>    Also note that packages installed in `user-contrib` or accessed through `COQPATH` are
   made available as if with a `-Q` option.  If you create a project outside of these
   using `-R` in `_CoqProject` and then later put in one of these locations,
   :cmd:`Require` commands in your code that depend on the `-R` behavior will fail in
   the new location.

Reply:
>Also, the following may also be incorrect--do you guys know? Are partially qualified names resolved at compile time or run time?

I'm not sure what you mean by "run time" (my default interpretation would be the same as "compile time"). If you're asking what's in .vo files then indeed they're fully qualified.

Reply:
Perhaps I should have said "load time".  Just carrying over programming language terminology imprecisely.